syntax = "proto3"; // 声明proto版本

package fixbug; // 声明代码所在的包（对于C++来说，就是namespace）

// 定义下面的选项，表示生成service服务代码和rpc方法描述，默认不生成
option cc_generic_services = true;

// 定义登录消息类型 name pwd
message LoginRequest {
    bytes name = 1;
    bytes pwd = 2;
    map<int32, bytes> test = 3;

}

// 定义相应登录消息类型 resultcode data
message ResultCode {
    int32 errcode = 1;
    bytes errmsg = 2;
    bool success = 3;
}

message LoginResponse {
    ResultCode result = 1;
    GetFriendListsRequest data = 2;
    bool success = 3;
}

message GetFriendListsRequest {
    uint32 userid = 1;
}

message User{
    bytes name = 1;
    uint32 age = 2;
    enum Sex{
        MAN = 0;
        WOMAN = 1;
    }
    Sex sex = 3;
}

message GetFriendListsResponse {
    ResultCode result = 1;
    repeated User friend_list = 2; // 定义了一个列表类型
}

// 在protobuf中定义服务
service UserServiceRpc {
    rpc Login(LoginRequest) returns(LoginResponse);
    rpc GetFriendLists(GetFriendListsRequest) returns(GetFriendListsResponse);
}